### PR TITLE
fixed steps load

### PR DESCRIPTION
**Pull request by monik182**

### CODE REVIEW

**Review by darthtrevino — APPROVED** [review]
Just some questions

File: javascript/core/src/engine/GraphManager.ts [hunk 1]
```diff
@@ -109,13 +110,16 @@ export class GraphManager {
...
+			return
+		}
 		this._workflow.clear()
 		// todo: add graph clear
 		this.graph.nodes.forEach(id => this._graph.remove(id))
 
 		// if a new workflow is injected, sync it into the graph
 		if (workflow != null) {
-			this._workflow = workflow
+			this._workflow = workflow.clone()
```
[the remark this answers is not in the document]
Comment (darthtrevino) [reply javascript/core/src/engine/GraphManager.ts:122]: I think the idea is 'reset to this workflow'

[the remark this answers is not in the document]
Comment (natoverse) [reply javascript/core/src/engine/GraphManager.ts:122]: gotcha

File: javascript/core/src/engine/Workflow.ts [hunk 2]
```diff
@@ -147,7 +156,7 @@ export class Workflow {
 			output.push({ ...binding, name })
 		}
 		return {
-			$schema: `https://microsoft.github.io/datashaper/schema/workflow/v1.json`,
+			$schema: `https://microsoft.github.io/datashaper/schema/workflow/v3.json`,
```
[the remark this answers is not in the document]
Comment (monik182) [reply javascript/core/src/engine/Workflow.ts:159]: We're already in v3 of the workflow schema, so I that's I updated it
image

File: javascript/react/src/components/ManageWorkflow.hooks.tsx [hunk 3]
```diff
@@ -21,25 +21,21 @@ import {
...
+	onSave?: (step: Step, output: string | undefined, index?: number) => void,
 ): (_step: Step) => void {
 	const createTableName = useCreateTableName(graph)
-	const argsDuplicator = useArgsDuplicator()
 
 	return useCallback(
 		(step: Step) => {
 			const outputTable = table ?? graph?.latestForNodeId(step.id)?.table
+			const clonedStep = cloneStep(step, outputTable?.columnNames())
+			clonedStep.id = ''
```
[the remark this answers is not in the document]
Comment (monik182) [reply javascript/react/src/components/ManageWorkflow.hooks.tsx:32]: It clears the id from the duplicated step and forces it to assign it a new one.

File: javascript/core/src/engine/Workflow.ts [hunk 4]
```diff
@@ -47,6 +48,14 @@ export class Workflow {
 		})
 	}
 
+	public clone(): Workflow {
+		const clone = new Workflow()
+		clone._steps = this._steps.map(x => cloneStep(x))
```
[the remark this answers is not in the document]
Comment (monik182) [reply javascript/core/src/engine/Workflow.ts:53]: That method receives a 2nd optional argument and when building the project I get a type error 
image

**Review by dayesouza — APPROVED** [review]

**Pull request merged**